src/CMakeLists.txt

```diff
@@ -295,6 +295,7 @@ if(WITH_SEASTAR)
     endif()
   endif()
   set(Seastar_CXX_FLAGS "-Wno-error;-Wno-sign-compare;-Wno-attributes;-Wno-pessimizing-move;-Wno-address-of-packed-member" CACHE STRING "" FORCE)
+  include_directories(seastar INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/c-ares")
```

Comment:
could you share the error message? in seastar's cmake file, i found
```cmake
target_link_libraries (seastar
  PUBLIC
    Boost::boost
    Boost::program_options
    Boost::thread
    c-ares::c-ares
    cryptopp::cryptopp
# ...
```
in in theory, seastar should expose c-ares' include directory to the targets linked against it already.

Reply:
the error message is as follows:
```
In file included from /home/xuxuehan/ceph/src/seastar/src/net/dns.cc:25:
/home/xuxuehan/ceph/src/c-ares/ares.h:22:10: fatal error: ares_build.h: No such file or directory
 #include "ares_build.h"    /* c-ares build definitions */
          ^~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [src/seastar/CMakeFiles/seastar.dir/build.make:636: src/seastar/CMakeFiles/seastar.dir/src/net/dns.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

Reply:
ares_build.h is in the build/src/c-ares directory

Reply:
@xxhdx1985126 so, in your case, it was seastar who failed to find c-ares' header file. i don't think your change helps. because

1. it's wrong. per https://cmake.org/cmake/help/latest/command/include_directories.html, there is no such a function signature of `include_directory(<target> INTERFACE <path>)`
1. it does not change how seastar search for the header files.

could you help test and review #33256 instead?

also i'd suggest you install `c-ares-devel` and other dependencies from the CentOS/EPEL repo.

Reply:
Um...this patch works for me, however, I do think your patch is better:-)

Reply:
the reason why it works is that it practically adds 3 include directories globally

- "seastar"
- "INTERFACE"
- "${CMAKE_CURRENT_BINARY_DIR}/c-ares"

and since "${CMAKE_CURRENT_BINARY_DIR}/c-ares" is also added, seastar will be able to find `ares_build.h`.